Framework/Automation.rb

```diff
@@ -0,0 +1,195 @@
+        tests_object.send_text(search_term, inputs)
+        tests_object.get_element_by_xpath(search_bar_xpath).click
+        sleep(15)
+        tests_object.close_popup(popup,"//*[@id='form-body-main']/div[1]/div/div")
+        sleep(5)
+        filter=tests_object.get_elements_by_class("snize-product-filters-list")[0]
+        items=tests_object.get_element_by_tag("li",filter)[0]
+        puts tests_object.get_element_by_xpath("//*[@id='snize_filters_block_snize_facet1']/div").click
+        puts "end with automation of add to cart module"
+        sleep(10)
+        tests_object.navigate_to_and_maximize("https://sublimelife.in/")
+    end    
+
+    def add_to_cart
+        puts "started with automation of add to cart module"
+        tests_object.scroll(2000)
+        container=tests_object.get_elements_by_class("slick-track")[2]
+        buttons=tests_object.get_element_by_tag("button",container)
+        puts  buttons[0].click
+        sleep(5)
+        c=tests_object.get_element_by_id("QuickView-martiderm-dsp-bright")
+        f=tests_object.get_element_by_tag("form",c)[0]
+        b=tests_object.get_element_by_tag("button",f);
+        b[1].click()
+        sleep(3)
+        b[2].click
+        sleep(10)
+        tests_object.get_element_by_xpath("//*[@id='ajaxCartForm']/div[3]/a").click()
+        sleep(5)
+        puts "finished with add_to_cart module"
+        sleep(10)
+        tests_object.navigate_to_and_maximize("https://sublimelife.in/")
+    end
+
+    def checkout_payment
+        puts "started testing the checkout module"
+        tests_object.scroll(2000)
+        container=tests_object.get_elements_by_class("slick-track")[2]
+        buttons=tests_object.get_element_by_tag("button",container)
+        puts  buttons[0].click
+        sleep(5)
+        c=tests_object.get_element_by_id("QuickView-martiderm-dsp-bright")
+        f=tests_object.get_element_by_tag("form",c)[0]
+        b=tests_object.get_element_by_tag("button",f);
+        b[1].click()
+        sleep(3)
+        b[2].click
+        sleep(5)
+        puts "[+]cart updated"
+        tests_object.get_element_by_xpath("//*[@id='ajaxCartForm']/div[3]/a").click()
+        sleep(5)
+        tests_object.navigate_to_and_maximize("https://sublimelife.in/")
+        sleep(5)
+        cart=tests_object.get_element_by_xpath("//*[@id='shopify-section-header']/div/header/div/div/div[4]/ul/li[2]/a").click
+        sleep(5)
+        tests_object.get_element_by_xpath("//*[@id='ajaxCartForm']/div[3]/a").click()
+        sleep(2)
+        tests_object.get_element_by_xpath("//*[@id='chck1']/div/div/div[2]/button").click()
+        sleep(5)
+        puts "[+]filling in deleievry address"
+        email=tests_object.get_element_by_id("email")
+        tests_object.send_text("test@test.com",email)
+        tf1=tests_object.get_element_by_id("TextField1")
+        tests_object.send_text("test",tf1)
+        tf2=tests_object.get_element_by_id("TextField2")
```

Comment:
Variable names should be descriptive of the type of data it holds. Rename variables appropriately